presto-spi/src/main/java/com/facebook/presto/spi/tracing/TracerProvider.java

```diff
@@ -15,9 +15,12 @@
 
 public interface TracerProvider
 {
+    String getName();
+
+    String getTracerType();
     /**
      *
      * @return A @Tracer that should be kept throughout the whole duration of tracing.
      */
-    Tracer getNewTracer();
+    Tracer getNewTracer(String traceToken, String contextPropagator, String propagatedContext);
```

Comment:
I created the w3c / b3 headers because tracing libraries such as Datadog `ddtrace` will automatically inject these headers and the dev does no additional work. I think using `PRESTO_TRACE_TOKEN` would oppose the standard context propagation convention and cause developers some confusion if expecting context propagation support. So I think having the new headers is still important. However, I can make it work with the same tracer interface.